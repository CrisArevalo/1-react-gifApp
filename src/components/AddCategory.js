import React, { useState } from "react";
import { PropTypes } from "prop-types";
import "./AddCategoryStyles.scss";

export const AddCategory = ({ setTags }) => {
    const [value, setValue] = useState("");

    const handleValueChange = (e) => {
        setValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if (value.trim().length > 2) {
            setTags((tags) => [value, ...tags]);
            setValue("");
        }
    };

    return (
        <div className="wrapper">
            <form onSubmit={handleSubmit}>
                <h2 className="Title">GifExpertApp</h2>
                <div className="Input">
                    <input
                        type="input"
                        value={value}
                        onChange={handleValueChange}
                        placeholder="Introduce una tematica..."
                        className="Input-text"
                        id="input"
                    />
                    <label className="Input-label">Gif input</label>
                </div>
            </form>
        </div>
    );
};

AddCategory.propTypes = {
    setTags: PropTypes.func.isRequired,
};
