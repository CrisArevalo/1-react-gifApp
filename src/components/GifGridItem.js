import React from "react";
import PropTypes from "prop-types";

export const GifGridItem = ({ url, title, id }) => {
    //En los parametros lo vuelvo a desestructurar.

    return (
        <>
            <img
                className="GifGrid animate__animated animate__bounce animate__bounceInRight"
                src={url}
                alt={id}
            />
        </>
    );
};

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
};
