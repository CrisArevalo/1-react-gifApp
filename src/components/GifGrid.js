import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ tags }) => {
    //"tags" solo es la categoria del inputs
    const { data } = useFetchGifs(tags);

    return (
        <>
            <h3>{tags}</h3>
            {data.map(({ id, title, url }) => {
                return <GifGridItem key={id} url={url} title={title} />;
            })}
        </>
    );
};

GifGrid.propTypes = {
    tags: PropTypes.string.isRequired,
};
