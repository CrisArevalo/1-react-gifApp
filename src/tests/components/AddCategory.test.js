import React from "react";
import PropTypes from "prop-types";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Tests en el <AddCategory/>", () => {
    const mock = jest.fn(); //Funcion vacia que ofrece Jest
    let wrapper = shallow(<AddCategory setTags={mock} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setTags={mock} />);
    });

    // ============ TESTS ============ //
    /*
    test("Test de Snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Test en el input", () => {
        const input = wrapper.find("input");
        const value = "Hello Me";

        input.simulate("change", { target: { value } });
    });

    test("Test del submit", () => {
        wrapper.find("form").simulate("submit", { preventDefault() {} });
        // Simulando el envio de datos al submit.

        expect(mock).toHaveBeenCalled();
    });

    test("Ejercicio 1", () => {
        const value = "Hello Me";

        // 1. simular el inputChange
        wrapper.find("input").simulate("change", { target: { value } });
        // 2. simular el submit
        wrapper.find("form").simulate("submit", { preventDefault() {} });
        // 3. mock debe ser llamado
        expect(mock).toHaveBeenCalled(); //toHaveBeenCalledTimes(1) <- Otra forma
        // 4. el valor del input debe estar "" (vacio).
        expect(wrapper.find("input").prop("value")).toBe("");
    });
    */
});
