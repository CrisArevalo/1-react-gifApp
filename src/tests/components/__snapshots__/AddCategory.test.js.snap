// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests en el <AddCategory/> Test de Snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
